fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java

```diff
@@ -0,0 +1,391 @@
+        return hudiProperties.get(HUDI_TABLE_INPUT_FORMAT);
+    }
+
+    public String getHudiMatadataEnable() {
+        return hudiProperties.get(HUDI_MATADATA_ENABLE);
+    }
+
+    private void validate(Map<String, String> properties) throws DdlException {
+        if (properties == null) {
+            throw new DdlException("Please set properties of iceberg table, they are: database, table.");
+        }
+
+        Map<String, String> copiedProps = Maps.newHashMap(properties);
+        this.db = copiedProps.get(HUDI_DB);
+        if (Strings.isNullOrEmpty(db)) {
+            throw new DdlException(String.format(PROPERTY_MISSING_MSG, HUDI_DB, HUDI_DB));
+        }
+        copiedProps.remove(HUDI_DB);
+
+        this.table = copiedProps.get(HUDI_TABLE);
+        if (Strings.isNullOrEmpty(table)) {
+            throw new DdlException(String.format(PROPERTY_MISSING_MSG, HUDI_TABLE, HUDI_TABLE));
+        }
+        copiedProps.remove(HUDI_TABLE);
+
+        String hudiMetaDataEnable = copiedProps.get(HUDI_MATADATA_ENABLE);
+        if (!Strings.isNullOrEmpty(hudiMetaDataEnable)) {
+            copiedProps.remove(HUDI_MATADATA_ENABLE);
+            hudiProperties.put(HUDI_MATADATA_ENABLE, hudiMetaDataEnable);
+        } else {
+            hudiProperties.put(HUDI_MATADATA_ENABLE, "false");
+        }
+
+        String resourceName = copiedProps.get(HUDI_RESOURCE);
+        Resource resource = Catalog.getCurrentCatalog().getResourceMgr().getResource(resourceName);
+        HudiResource hudiResource = (HudiResource) resource;
+        if (hudiResource == null) {
+            throw new DdlException("hudi resource [" + resourceName + "] not exists");
+        }
+        if (hudiResource.getType() != Resource.ResourceType.HUDI) {
+            throw new DdlException("resource [" + resourceName + "] is not hudi resource");
+        }
+        hudiProperties.put(HUDI_METASTORE_URIS, hudiResource.getHiveMetastoreURIs());
+
+        org.apache.hadoop.hive.metastore.api.Table metastoreTable = Catalog.getCurrentCatalog().getHiveRepository()
+                .getTable(resourceName, this.db, this.table);
+        String metastoreTableType = metastoreTable.getTableType();
+        if (metastoreTableType == null) {
+            throw new DdlException("Unknown metastore table type.");
+        }
+        switch (metastoreTableType) {
+            case "VIRTUAL_VIEW":
+                throw new DdlException("VIRTUAL_VIEW table is not supported.");
+            case "EXTERNAL_TABLE":
+            case "MANAGED_TABLE":
+                break;
+            default:
+                throw new DdlException("unsupported hudi table type [" + metastoreTableType + "].");
+        }
+
+        this.resourceName = resourceName;
+        copiedProps.remove(HUDI_RESOURCE);
+
+        String hudiBasePath = metastoreTable.getSd().getLocation();
+        if (!Strings.isNullOrEmpty(hudiBasePath)) {
+            hudiProperties.put(HUDI_BASE_PATH, hudiBasePath);
```

Comment:
Maybe we should check this property since base path is necessary for hudi table?

Reply:
you mean existence or something else of that?

Reply:
return error if null or empty during create table